lib/util.js

```diff
@@ -6,9 +6,17 @@ const Debug = require('vm').runInDebugContext('Debug');
 const formatRegExp = /%[sdj%]/g;
 exports.format = function(f) {
   if (typeof f !== 'string') {
+    var i = 0;
+    var len = arguments.length;
     var objects = [];
-    for (var i = 0; i < arguments.length; i++) {
-      objects.push(inspect(arguments[i]));
+    for (var x = arguments[i]; i < len; x = arguments[++i]) {
```

Comment:
That loop is identical to the existing code: https://github.com/nodejs/node/blob/master/lib/util.js#L38

I wouldn't write it that way, either, but I'd rather not introduce stylistic variations, or at least, follow up with a commit to improve both for loops simultaneously.